internal/brokerstore/brokerstore.go

```diff
@@ -81,55 +79,54 @@ func NewBrokerStore(variant string, config *messagebroker.BrokerConfig) (IBroker
 	}
 
 	return &BrokerStore{
-		variant: variant,
-		bConfig: config,
+		variant:       variant,
+		bConfig:       config,
+		partitionLock: partitionlocker.NewPartitionLocker(&sync.Mutex{}),
 	}, nil
 }
 
 // GetOrCreateConsumer returns for an existing consumer instance, if available returns that else creates as new instance
-func (b *BrokerStore) GetOrCreateConsumer(ctx context.Context, op messagebroker.ConsumerClientOptions) (messagebroker.Consumer, error) {
-	key := NewKey(b.variant, op.Partition)
-
-	if consumer, exists := b.consumerMap.Load(key.String()); exists {
+func (b *BrokerStore) GetOrCreateConsumer(ctx context.Context, id string, op messagebroker.ConsumerClientOptions) (messagebroker.Consumer, error) {
```

Comment:
should we just rename this to GetConsumer, GetProducer ?

Reply:
Done